virtool/samples/data.py

```diff
@@ -158,29 +158,32 @@ async def find(
         )
 
     async def get(self, sample_id: str) -> Sample:
-        document = await self._db.samples.find_one({"_id": sample_id})
-
-        if not document:
-            raise ResourceNotFoundError
+        async for document in self._db.samples.aggregate(
```

Comment:
Since you know that you should only ever get one document I would be tempted to simply call `to_list` on the mongo cursor returned by `self._db.samples.aggregate` rather than using an async for loop. Then if the list is empty throw the  `ResoruceNotFound` error.